cg/meta/demultiplex/status_db_storage_functions.py

```diff
@@ -25,28 +25,33 @@
 def store_flow_cell_data_in_status_db(
     parsed_flow_cell: FlowCellDirectoryData, store: Store
 ) -> None:
-    """Create flow cell from the parsed and validated flow cell data."""
-    if not store.get_flow_cell_by_name(flow_cell_name=parsed_flow_cell.id):
+    """
+    Create flow cell from the parsed and validated flow cell data.
+    And add the samples on the flow cell to the model.
+    """
+    flow_cell: Flowcell = store.get_flow_cell_by_name(flow_cell_name=parsed_flow_cell.id)
+    if not flow_cell:
         flow_cell: Flowcell = Flowcell(
             name=parsed_flow_cell.id,
             sequencer_type=parsed_flow_cell.sequencer_type,
             sequencer_name=parsed_flow_cell.machine_name,
             sequenced_at=parsed_flow_cell.run_date,
         )
-        sample_internal_ids = get_sample_internal_ids_from_sample_sheet(
-            sample_sheet_path=parsed_flow_cell.sample_sheet_path,
-            flow_cell_sample_type=parsed_flow_cell.sample_type,
-        )
-        flow_cell = add_samples_to_flow_cell_in_status_db(
-            flow_cell=flow_cell,
-            sample_internal_ids=sample_internal_ids,
-            store=store,
-        )
-        store.session.add(flow_cell)
-        store.session.commit()
         LOG.info(f"Flow cell added to status db: {parsed_flow_cell.id}.")
     else:
-        LOG.info(f"Flow cell already exists in status db: {parsed_flow_cell.id}. Skipping.")
+        LOG.info(f"Flow cell already exists in status db: {parsed_flow_cell.id}.")
+    sample_internal_ids = get_sample_internal_ids_from_sample_sheet(
+        sample_sheet_path=parsed_flow_cell.sample_sheet_path,
+        flow_cell_sample_type=parsed_flow_cell.sample_type,
+    )
+    flow_cell: Flowcell = add_samples_to_flow_cell_in_status_db(
```

Comment:
It is not necessary to return the flow cell object from `add_samples_to_flow_cell_in_status_db`, but you decide if it is more readable

Reply:
We already have a reference to the flow cell object that is mutated in `add_samples_to_flow_cell_in_status_db` which means we can just commit the changes to the object using that reference